x/upgrade/client/rest/query.go

```diff
@@ -0,0 +1,41 @@
+package rest
+
+import (
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	upgrade "github.com/cosmos/cosmos-sdk/x/upgrade/internal/types"
+	"github.com/gorilla/mux"
+)
+
+// RegisterRoutes registers REST routes for the upgrade module under the path specified by routeName.
+func RegisterRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Codec, storeName string) {
+	r.HandleFunc("/"+upgrade.ModuleName, getUpgradePlanHandler(cdc, cliCtx, storeName)).Methods("GET")
+}
+
+func getUpgradePlanHandler(cdc *codec.Codec, cliCtx context.CLIContext, storeName string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, request *http.Request) {
+		// ignore height for now
+		res, _, err := cliCtx.QueryStore(upgrade.PlanKey(), storeName)
```

Comment:
ok, so here you are directly querying the Store instead of using the `Querier` (_i.e_ by using `cliCtx.Query`/ `cliCtx.QueryWithData`). Could you use the later or open an issue to address it later? I noticed the `TODO` on `module.go`.

Reply:
Yeah, I didn't even know about the other approach until I tried to wire up the module and saw this as null and dug into it.

Yes, this is a TODO, will resolve tomrorow.